lib/src/revset.rs

```diff
@@ -1885,15 +1885,14 @@ fn resolve_remote_bookmark(repo: &dyn Repo, name: &str, remote: &str) -> Option<
         .then(|| target.added_ids().cloned().collect())
 }
 
-fn all_bookmark_symbols(
+fn all_formatted_bookmark_symbols(
     repo: &dyn Repo,
     include_synced_remotes: bool,
 ) -> impl Iterator<Item = String> + '_ {
     let view = repo.view();
     view.bookmarks().flat_map(move |(name, bookmark_target)| {
-        // Remote bookmark "x"@"y" may conflict with local "x@y" in unquoted form.
```

Comment:
Can the `.dedup()` in `collect_similar()` be dropped now? I would expect the candidates passed to that function to not have duplicates, and I'm guessing it did only because of the conflict fixed by this patch.

Reply:
Hmm, yes. Templater could pass duplicated `"self"` entries in very rare cases, but they'll be deduplicated later.

Removed `.dedup()`.